### PR TITLE
Add doc test for YAML configuration

### PR DESCRIPTION
**Pull request by dadada**
This adds a doc test and some documentation for the configuration file.

### CODE REVIEW

File: hypervisor/src/hypervisor/config.rs [hunk 1]
```diff
@@ -1,3 +1,52 @@
+//! Configuration for a653rs-linux-hypervisor.
+//!
+//! A configuration contains information about the partition schedule.
+//! Currently, only a single schedule is supported. Each schedule's major frame
+//! (MAF) has a fixed duration and number of slots. Each slot is occupied by
+//! at most one partition that executes during the slot. A slot has a fixed
+//! duration and offset inside the MAF. A partition may occupy multiple slots
+//! inside the schedule, in which case it may be repeated using the `period`
+//! parameter.
```
Comment (sevenautumns) [inline hypervisor/src/hypervisor/config.rs:9]: Why are you doing line-breaks manually

File: hypervisor/src/hypervisor/config.rs [hunk 2]
```diff
@@ -1,3 +1,52 @@
...
+//! the partition schedule.
+//!
+//! Partitions can communicate using channels (Sampling and Queuing). The name
+//! of the ports by which a partition can access a channel is the same for all
+//! attached partitions.
+//!
+//! ```rust
+//! use a653rs_linux_hypervisor::hypervisor::config::Config;
+//!
+//! let yaml = "
```
Comment (sevenautumns) [inline hypervisor/src/hypervisor/config.rs:22]: Unnecessary parts of a doc test could be hidden

https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example

File: hypervisor/src/hypervisor/config.rs [hunk 3]
```diff
@@ -31,6 +80,7 @@ pub struct Partition {
     pub duration: Duration,
     #[serde(with = "humantime_serde")]
     pub offset: Duration,
+    /// Repetition interval of the slice inside the MAF.
```
Comment (sevenautumns) [inline hypervisor/src/hypervisor/config.rs:83]: also a the MAF must be cleanly dividable by this `period`

Comment (dadada) [reply hypervisor/src/hypervisor/config.rs:83]: Good point!

File: hypervisor/src/hypervisor/config.rs (hunk 1 above)
Comment (dadada) [reply hypervisor/src/hypervisor/config.rs:9]: What would be the alternative? 80 characters seems to be the default. I'd like to keep the documentation readable, regardless of the editor. Looked here for reference on how to do this. https://doc.rust-lang.org/src/std/lib.rs.html#1-686

Comment (sevenautumns) [reply hypervisor/src/hypervisor/config.rs:9]: acceptable

File: hypervisor/src/hypervisor/config.rs [hunk 4]
```diff
@@ -1,3 +1,49 @@
+//! Configuration for a653rs-linux-hypervisor.
+//!
+//! A configuration contains information about the partition schedule.
+//! Currently, only a single schedule is supported. Each schedule's major frame
+//! (MAF) has a fixed duration and number of slots. Each slot is occupied by at
+//! most one partition that executes during the slot. A slot has a fixed
+//! duration and offset inside the MAF. A partition may occupy multiple slots
+//! inside the schedule, in which case it may be repeated using the `period`
+//! parameter. Also a the MAF must be cleanly dividable by this period.
```
Comment (sevenautumns) [inline hypervisor/src/hypervisor/config.rs:9]: You copied my typo

**Review by sevenautumns — APPROVED** [review]

**Pull request merged**